### ISSUE TITLE
bug(YouTube/Hide feed components): `Hide carousel shelf` setting sometimes hides the library shelf

### ISSUE THREAD

**Issue by Spaceshiprace** [inotia00/ReVanced_Extended#1970]
### Type

Error at runtime

### Tools used

RVX Manager

### Application

YT 19.16.39

### Bug description

Please, I need help I just don't know why the playlist keeps disappear.. like i just opens the app search for something or watch a video then when i go to library this happens (No watch later playlist or any other playlist appears)
![Screenshot_2024-05-09-14-46-07-795_app rvx android youtube_1](https://github.com/inotia00/ReVanced_Extended/assets/156727749/d7af4f75-0043-419a-a9ff-4904e2ae54bd)

### Error logs

```shell
non
```

### Solution

idk

### Device Environment

Android

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
It is probably caused by the following settings:
```
Settings > ReVanced Extended > Feed > Hide carousel shelf
```

Please disable this setting until the issue is resolved.

**Comment by inotia00** [comment]
fixed in [revanced-patches-4.8.0-dev.2](https://github.com/inotia00/revanced-patches/releases/tag/v4.8.0-dev.2)
